carma_guidance_plugins/src/control_plugin.cpp

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (C) 2022 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include <functional>
+#include "carma_guidance_plugins/control_plugin.hpp"
+
+
+namespace carma_guidance_plugins
+{
+  namespace std_ph = std::placeholders;
+
+  ControlPlugin::ControlPlugin(const rclcpp::NodeOptions &options)
+      : PluginBaseNode(options)
+  {}
+
+  std::string ControlPlugin::get_capability()
+  {
+    return "control/trajectory_control";
+  }
+
+  uint8_t ControlPlugin::get_type() 
+  {
+    return carma_planning_msgs::msg::Plugin::CONTROL; 
+  }
+
+  void ControlPlugin::current_pose_callback(geometry_msgs::msg::PoseStamped::UniquePtr msg)
```

Comment:
Does it make sense to add logging to these methods, just to have it? Should we consider adding it to the state transitions as well? Should be easy enough to wrap the user-defined callback with a higher-level one that simply logs the state transition then calls the user's callback, I think.

Reply:
I believe, the state transitions already have logs from the carma base node. I can add debug logs to the callbacks. 

Can you expand on your last sentence. I'm not quite clear on what you're asking.

Reply:
If the state transitions are logged then that's really what I was looking for, just to ensure we have records.

RE: the last sentence. I think I was getting a few wires crossed mentally, I was thinking for some reason these callbacks would be user defined (e.g., something like a `set_current_pose_cb(Callback cb)` kind of a pattern. That we could still ensure those were logged by wrapping them in a callback of our own (which just logs the message and then invokes the user-defined callback) before registering that callback with the subscriber. Doesn't particularly matter here, that's not the pattern we're using.

Digression aside, adding some debug level logs here wouldn't hurt.

Reply:
Added debug logs